SchemaMigration/COSchemaMigration.h

```diff
@@ -0,0 +1,417 @@
+ * };
+ *
+ * COModelElementMove *move = [COModelElementMove new];
+ *
+ * move.name = @"Person";
+ * move.packageName = @"com.company.AddressBook";
+ * move.packageVersion = 5;
+ * migration.entityMoves = S(move);
+ *
+ * [COSchemaMigration registerMigration: migration];
+ * </example>
+ */
+@interface COSchemaMigration : NSObject
+{
+	@private
+	NSString *_packageName;
+	int64_t _destinationVersion;
+	COMigrationBlock _migrationBlock;
+	__weak COSchemaMigrationDriver *migrationDriver;
+	NSDictionary *_dependentSourceVersionsByDomain;
+}
+
+
+/** @taskunit Schema Migration Registration */
+
+
+/**
+ * Registers a schema migration.
+ *
+ * See +migrationForVersion:.
+ */
++ (void)registerMigration: (COSchemaMigration *)migration;
+/**
+ * Returns a schema migration to update store items to a newer schema version.
+ *
+ * The returned migration applies to store items one version behind, items using 
+ * an older schema version must be first migrated to <em>version - 1</em>.
+ *
+ * See +registerMigration: and -destinationVersion.
+ */
++ (COSchemaMigration *)migrationForPackageName: (NSString *)package
+                       destinationVersion: (NSInteger)version;
+/**
+ * Returns a dictionary that contains dependent migrations to be run before a 
+ * specific migration. 
+ *
+ * The dictionary keys are key-value pair, where -[ETKeyValue key] returns the 
+ * package name and -[ETKeyValue value] returns the destination version.
+ */
++ (NSDictionary *)dependencies;
+
+/**
+ * Returns an array of all registered migrations.
+ */
++ (NSArray *)migrations;
+
+
+/** @taskunit Migrating to Future Versions */
+
+
++ (NSArray *)migrateItems: (NSArray *)storeItems
+withModelDescriptionRepository: (ETModelDescriptionRepository *)repo;
+
+
+/** @taskunit Targeted Versions */
+
+
+/**
+ * The package name that must correspond to a package name in the metamodel.
+ *
+ * See -[ETPackageDescription name] and -[COCommitDescriptor domain].
+ */
+@property (nonatomic, copy) NSString *packageName;
+/**
+ * The new schema version.
+ *
+ * By default, returns 0.
+ *
+ * See -sourceVersion.
+ */
+@property (nonatomic, assign) int64_t destinationVersion;
+/**
+ * The old schema version, one version behind -destinationVersion.
+ *
+ * By default, returns 0.
+ *
+ * A source schema version is included in each item passed to -migrateItems:,
+ * both values must match to carry the migration out.
+ */
+@property (nonatomic, readonly) int64_t sourceVersion;
+/**
+ * A migration block called by -migrateItems:.
+ *
+ * You can put the migration logic in this block, rather than creating a 
+ * subclass and overriding -migrateItems:.
+ */
+@property (nonatomic, copy) COMigrationBlock migrationBlock;
+/**
+ * Domains depended on by the -sourceVersion of -packageName, along with
+ * their versions.
+ *
+ * This acts as a snapshot of the necessary parts of the metamodel at 
+ * -sourceVersion.
+ */
+@property (nonatomic, copy) NSDictionary *dependentSourceVersionsByDomain;
```

Comment:
This property should probably be renamed to 'dependentSourceVersionsByPackageName', and the doc updated to 'Package name depended on …'